okonomiyaki/file_formats/tests/test_egg_file_format.py

```diff
@@ -286,6 +287,37 @@ def it():
         with zipfile2.ZipFile(egg_path, "r") as fp:
             self.assertEqual(set(fp.namelist()), set(r_files))
 
+    def test_files_are_readable(self):
+        # Given
+        metadata = self._create_fake_metadata()
+        r_prefix = os.path.join(self.d, "prefix")
+
+        # When
+        with EggBuilder(metadata, cwd=self.d) as fp:
+            pass
+
+        egg_path = os.path.join(self.d, "Qt_debug-4.8.6-1.egg")
+
+        with zipfile2.ZipFile(egg_path, "r") as fp:
+            fp.extractall(
+                r_prefix, preserve_permissions=zipfile2.PERMS_PRESERVE_SAFE
+            )
+
+        # Then
+        # Files are readable for everybody
+        for archive in (
+            "EGG-INFO/PKG-INFO",
+            "EGG-INFO/spec/depend",
+            "EGG-INFO/spec/summary",
+        ):
+            pkg_info = os.path.join(r_prefix, archive)
+            self.assertTrue(os.path.exists(pkg_info))
+            perms = stat.S_IMODE(os.stat(pkg_info).st_mode)
+            if sys.platform == "win32":
+                self.assertEqual(perms, 0o666)
+            else:
+                self.assertEqual(perms, 0o644)
+
```

Comment:
Seems trivial, but can you add tests for the error handling when `archive_name[-1] == "/"` and/or when `isinstance(archive_name, zipfile.ZipInfo) == True`.